src/sagemaker/jumpstart/curated_hub/jumpstart_curated_public_hub.py

```diff
@@ -0,0 +1,362 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module provides the JumpStart Curated Hub class."""
+from __future__ import absolute_import
+
+import json
+from concurrent import futures
+from typing import List, Optional, Tuple, Dict
+
+import boto3
+from botocore.client import ClientError
+
+from sagemaker.jumpstart.curated_hub.content_copy import ContentCopier
+from sagemaker.jumpstart.curated_hub.hub_client import CuratedHubClient
+from sagemaker.jumpstart.curated_hub.model_document import ModelDocumentCreator
+from sagemaker.jumpstart.curated_hub.stsClient import StsClient
+from sagemaker.jumpstart.curated_hub.hub_model_specs.hub_model_specs import Dependency
+from sagemaker.jumpstart.curated_hub.accessors.public_hub_s3_accessor import (
+    PublicHubS3Accessor,
+)
+from sagemaker.jumpstart.curated_hub.accessors.curated_hub_s3_accessor import (
+    CuratedHubS3Accessor,
+)
+from sagemaker.jumpstart.curated_hub.utils import (
+    PublicModelId,
+    get_studio_model_metadata_map_from_region,
+)
+from sagemaker.jumpstart.enums import (
+    JumpStartScriptScope,
+)
+from sagemaker.jumpstart.types import JumpStartModelSpecs
+from sagemaker.jumpstart.utils import (
+    verify_model_region_and_return_specs,
+)
+from sagemaker.jumpstart.curated_hub.accessors.s3_object_reference import (
+    create_s3_object_reference_from_uri,
+)
+from sagemaker.jumpstart.curated_hub.utils import (
+    find_objects_under_prefix,
+)
+
+
+class JumpStartCuratedPublicHub:
+    """JumpStartCuratedPublicHub class.
+
+    This class helps users create a new curated hub.
+    If a hub already exists on the account, it will attempt to use that hub.
+    """
+
+    def __init__(
+        self,
+        curated_hub_name: str,
+        import_to_preexisting_hub: bool = False,
+        region: str = "us-west-2",
```

Comment:
can we use this constant? https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/jumpstart/constants.py#L130

Reply:
+1